### PR TITLE
Add a troubleshooting section to the developer's guide

### PR DESCRIPTION
**Pull request by xylar**
This will be an area for collecting common issues that developers run into and their likely solutions.

### CODE REVIEW

**Comment by xylar** [comment]
@altheaden, could you let me know if this describes the problem you ran into in a way that makes sense to you, and if the solution works for you?

File: docs/developers_guide/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+.. code-block:: none
+
+    Encountered problem while solving:
+      - nothing provides geos 3.5.* needed by cartopy-0.14.3-np110py27_4
+
+    ...
+
+    subprocess.CalledProcessError: ...
+
+Details of teh error may vary but the message indicates in some way that there
```
Comment (altheaden) [inline docs/developers_guide/troubleshooting.rst:31]: Typo - "teh" instead of "the"

**Comment by altheaden** [comment]
That sounds good to me! You could add a line about what `--recreate` does, compared to just updating, but it's not necessary for solving the problem. Maybe the explanation would be helpful, or maybe the troubleshooting guide should be left as simple as possible - I think that's a design decision.

File: docs/developers_guide/troubleshooting.rst (hunk 1 above)
Comment (xylar) [reply docs/developers_guide/troubleshooting.rst]: Fixed!

**Comment by xylar** [comment]
@altheaden, thanks a lot for taking a look.  I added a bit more text about what `--recreate` does.  Let me know if that looks good (and if you notice more typos!).

**Comment by altheaden** [comment]
@xylar everything looks good to me!

**Pull request merged**